src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -331,6 +331,16 @@ private IEnumerable<CustomNodeInfo> ScanNodeHeadersInDirectory(string dir, bool
         /// </summary>
         private void SetNodeInfo(CustomNodeInfo newInfo)
         {
+            var guids = NodeInfos.Where(x =>
+                        {
+                            return string.Compare(x.Value.Path, newInfo.Path, StringComparison.OrdinalIgnoreCase) == 0;
+                        }).Select(x => x.Key).ToList();
+
+            foreach (var guid in guids)
```

Comment:
`foreach` works for `IEnumerable`, therefore we don't need to convert the result to list, which memory allocation happens.